Discussion: Improved management of uptime kuma (#5728)
louislam/uptime-kuma: .github/PULL_REQUEST_TEMPLATE.md

We appreciate your patience and understanding as we continue to improve Uptime Kuma.

## **❗ Important: Please Avoid Unnecessary Pinging of Maintainers**

**We kindly ask users to refrain from pinging maintainers unless absolutely necessary. Pings should be reserved for critical issues or urgent matters that require immediate attention.**

- **Why**: To help maintainers focus on high-priority tasks, we kindly request that unnecessary or repeated pings be minimized. This will help ensure quicker responses for matters that truly need attention.
- **What This Means**: Non-urgent pings may be ignored or addressed later. Please assess the urgency of your request before pinging a maintainer.

## 📋 Overview

**Required**: Please fill out this section before submitting.

Provide a clear summary of the purpose and scope of this pull request:

- **What problem does this pull request address?**

  - Please provide a detailed explanation here.

- **Why is it necessary?**

  - Please provide a detailed explanation here.

- **What features or functionality does this pull request introduce or enhance?**

  - Please provide a detailed explanation here.

## 🔄 Changes

### 🛠️ Type of change

> **Required**: Please fill out this section before submitting.

Reviewer: Maybe hiding some of the explaination with markdown comments would make this less "wordy".
Alternatively, we should work with `<details` blocks to make this less long

Unsure what to hide.

Author: This has been updated.

> ## 🔄 Changes
> 
> ### 🛠️ Type of change
> 
> **Required**: Please fill out this section before submitting.
> 
> Please select all options that apply:
> 
> - [ ] 🐛 Bugfix (a non-breaking change that resolves an issue)
> - [ ] ✨ New feature (a non-breaking change that adds new functionality)
> - [ ] ⚠️ Breaking change (a fix or feature that alters existing functionality in a way that could cause issues)
> - [ ] 🎨 User Interface (UI) updates
> - [ ] 📄 New Documentation (addition of new documentation)
> - [ ] 📄 Documentation Update (modification of existing documentation)
> - [ ] 📄 Documentation Update Required (the change requires updates to related documentation)
> - [ ] 🔧 Other (please specify):
>   - Provide additional details here.
> 
> ## 🔗 Related Issues
> 
> **Optional**: Remove this section if not applicable.
> 
> Reference any related GitHub issues or tasks that this pull request addresses. Use proper issue links or numbers, for example:
> 
> - Relates to #issue-number
> - Resolves #issue-number
> - Fixes #issue-number

Reviewer: the PR template is much better, but also now much too long.
We need to cut here

```suggestion
## 🔗 Related Issues

**Optional**: Remove this section if not applicable.

Reference any related GitHub issues or tasks that this pull request addresses. Use issue links, for example:

- Relates to #issue-number
- Resolves #issue-number
```

Author: This has been updated.

- **What You Can Do**: If you're interested in contributing, reviewing open PRs by following our [Review Guidelines](REVIEW_GUIDELINES.md) or offering support to other users is greatly appreciated. All feature requests and PRs will be revisited once the suspension period is lifted.

We appreciate your patience and understanding as we continue to improve Uptime Kuma.

### 🚫 Please Avoid Unnecessary Pinging of Maintainers

**We kindly ask users to refrain from pinging maintainers unless absolutely necessary. Pings should be reserved for critical issues or urgent matters that require immediate attention.**

- **Why**: To help maintainers focus on high-priority tasks, we kindly request that unnecessary or repeated pings be minimized. This will help ensure quicker responses for matters that truly need attention.
- **What This Means**: Non-urgent pings may be ignored or addressed later. Please assess the urgency of your request before pinging a maintainer.

#### Guidelines for Pinging

- ✅ **Appropriate**: Critical issues or urgent matters requiring immediate attention.
- ❌ **Not Appropriate**: Repeated or non-urgent pings, which may be ignored or addressed later.

</p>
</details>

> ## 📋 Overview *

Reviewer: ```suggestion
## 📋 Overview
```

Provide a clear summary of the purpose and scope of this pull request:

- **What problem does this pull request address?**

  - Please provide a detailed explanation here.

> - **Why is it necessary?**
> 
>   - Please provide a detailed explanation here.
> 

Reviewer: Likely boils down to the same information. Not sure

```suggestion
```

- **What features or functionality does this pull request introduce or enhance?**

  - Please provide a detailed explanation here.

## 🔄 Changes

> ### 🛠️ Type of change *
> 
> Please select all options that apply:

Reviewer: ```suggestion
### 🛠️ Type of change

<!-- Please select all options that apply -->
```

- [ ] 🐛 Bugfix (a non-breaking change that resolves an issue)
- [ ] ✨ New feature (a non-breaking change that adds new functionality)
- [ ] ⚠️ Breaking change (a fix or feature that alters existing functionality in a way that could cause issues)
- [ ] 🎨 User Interface (UI) updates
- [ ] 📄 New Documentation (addition of new documentation)
- [ ] 📄 Documentation Update (modification of existing documentation)
- [ ] 📄 Documentation Update Required (the change requires updates to related documentation)
- [ ] 🔧 Other (please specify):
  - Provide additional details here.

## 🔗 Related Issues

> Please link any GitHub issues or tasks that this pull request addresses. Use the appropriate issue numbers or links.
> 
> <details><summary>Click to view more details:</summary>
> </p>
> 
> **Note**: Include only issues directly related to this PR. Remove any irrelevant reference.
> 
> - Relates to #issue-number
> - Resolves #issue-number
> - Fixes #issue-number
> 
> </p>
> </details>

Reviewer: ```suggestion
<!-- 
Please link any GitHub issues or tasks that this pull request addresses. Use the appropriate issue numbers or links.

**Note**: Include only issues directly related to this PR. Remove any irrelevant reference.
-->

- Relates to #issue-number
- Resolves #issue-number
- Fixes #issue-number
```

## 📄 Checklist *

> 
> Please select all options that apply:

Reviewer: ```suggestion

<!-- Please select all options that apply -->
```

- [ ] 🔍 My code adheres to the style guidelines of this project.
- [ ] ✅ I ran ESLint and other code linters for modified files.
- [ ] 🛠️ I have reviewed and tested my code.
- [ ] 📝 I have commented my code, especially in hard-to-understand areas (e.g., using JSDoc for methods).
- [ ] ⚠️ My changes generate no new warnings.
- [ ] 🤖 My code needed automated testing. I have added them (this is an optional task).
- [ ] 📄 Documentation updates are included (if applicable).
- [ ] 🔒 I have considered potential security impacts and mitigated risks.
- [ ] 🧰 Dependency updates are listed and explained.
- [ ] 📚 I have read and understood the [Pull Request guidelines](../CONTRIBUTING.md#recommended-pull-request-guideline).

## 📷 Screenshots or Visual Changes

> Please upload the image directly here by pasting it or dragging and dropping. Avoid using external image services as the image will be uploaded automatically.
> 
> <details><summary>Click here for more details:</summary>
> </p>
> 
> If this pull request introduces visual changes, please provide the following details:
> 
> - **UI Modifications**: Highlight any changes made to the user interface.
> - **Before & After**: Include screenshots or comparisons (if applicable).
> 
> | Event              | Before                | After                |
> | ------------------ | --------------------- | -------------------- |
> | `UP`               | ![Before](image-link) | ![After](image-link) |
> | `DOWN`             | ![Before](image-link) | ![After](image-link) |
> | Certificate-expiry | ![Before](image-link) | ![After](image-link) |
> | Testing            | ![Before](image-link) | ![After](image-link) |
> 
> </p>
> </details>

Reviewer: ```suggestion
<!--
Please upload the image directly here by pasting it or dragging and dropping. Avoid using external image services as the image will be uploaded automatically.

If this pull request introduces visual changes, please provide the following details.
If not, remove this section.
-->

- **UI Modifications**: Highlight any changes made to the user interface.
- **Before & After**: Include screenshots or comparisons (if applicable).

| Event              | Before                | After                |
| ------------------ | --------------------- | -------------------- |
| `UP`               | ![Before](image-link) | ![After](image-link) |
| `DOWN`             | ![Before](image-link) | ![After](image-link) |
| Certificate-expiry | ![Before](image-link) | ![After](image-link) |
| Testing            | ![Before](image-link) | ![After](image-link) |
```

## ℹ️ Additional Context

Provide any relevant details to assist reviewers in understanding the changes.

<details><summary>Click here for more details:</summary>
</p>

**Key Considerations**:

- **Design decisions** – Key choices or trade-offs made during development.
- **Alternative solutions** – Approaches considered but not implemented, along with reasons.
- **Relevant links** – Specifications, discussions, or resources that provide context.
- **Dependencies** – Related pull requests or issues that must be resolved before merging.
- **Additional context** – Any other details that may help reviewers understand the changes.

Provide details here

> ## 📄 Document Status
> 
> To streamline the review process, please list the filenames of documents currently being worked on. This helps reviewers focus on the files that need their attention.
> 
> ### ✅ Ready for Review
> 
> The following documents are complete and ready for review:
> 
> - `Specify filenames here`
> 
> ### 🔨 Work in Progress
> 
> The following documents are currently being worked on:
> 
> - `List ongoing documents here`
> 
> ### 💬 Requested Feedback
> 
> Your feedback is requested on the following documents:
> 
> - `Mention documents needing feedback here`
> 

Reviewer: I don't think that thes docs are enough usefull for the effort involved.
Lets strip it down to this one

```suggestion
## 💬 Requested Feedback

<!-- If a part of our docs is unclear, you are unsure how to do something/.. this is where we would appreciate your feedback -->

- `Mention documents needing feedback here`
```

> ## 📝 Additional Changes
> 
> Please provide any relevant details or context. (e.g., resolved Linter issues, code optimizations, or other adjustments):
> 
> - Describe additional changes here
> 
> </p>
> </details>

Reviewer: Unsure what we would recive here. If signal to noice is too low, nobody is reading this => lets remove it

```suggestion
```
